examples/sk_agent.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+@Time    : 2023/9/13 12:36
+@Author  : femto Zheng
+@File    : sk_agent.py
+"""
+import asyncio
+import os
+
+from semantic_kernel.core_skills import FileIOSkill, MathSkill, TextSkill, TimeSkill
+from semantic_kernel.planning import SequentialPlanner
+
+# from semantic_kernel.planning import SequentialPlanner
+from semantic_kernel.planning.action_planner.action_planner import ActionPlanner
+
+from metagpt.actions import BossRequirement
+from metagpt.roles.sk_agent import SkAgent
+from metagpt.schema import Message
+from metagpt.tools.search_engine import SkSearchEngine
+
+# Get the directory of the current file
+current_file_directory = os.path.dirname(os.path.abspath(__file__))
+# Construct the skills_directory by joining the parent directory and "skillss"
+skills_directory = os.path.join(current_file_directory, "..", "metagpt", "skills")
+# Normalize the path to ensure it's in the correct format
+skills_directory = os.path.normpath(skills_directory)
+
+
+async def main():
+    await basic_planner_example()
+    await action_planner_example()
+
+    # await sequential_planner_example()
+    # await basic_planner_web_search_example()
+
+
+async def basic_planner_example():
+    task = """
+    Tomorrow is Valentine's day. I need to come up with a few date ideas. She speaks French so write it in French.
+    Convert the text to uppercase"""
+    role = SkAgent()
+
+    # let's give the agent some skills
+    role.import_semantic_skill_from_directory(skills_directory, "SummarizeSkill")
```

Comment:
For a particular role, is it proper to make these skills as default build-in ones or pass them with `SkAgent(skills=[...])`. So we can only consider the role instance and `run`.

Reply:
for particular some specific role, we can just subclass SkAgent and hardcode its skills  in its constructor.
if you want to support skills=[...] in constructor, 
since we have both import_semantic_skill_from_directory and import_skill, these two have different signature,
what do you suggest the params look like. Also another problem is import_skill is using an object when registered,
```
role.import_skill(TextSkill(), "TextSkill")
```
let's say we instantiate some Action/Skill, that Action/Skill depend on role(like I discussed in WriteManyCodes),
```
role.import_skill(WriteManyCodes(role=self), "WriteCodeSkill")
```
then this line must happen after role's constructor